### PR TITLE
#300 - Correct usage of IntervalArithmetic

### LINKED ISSUE #300: Correct usage of IntervalArithmetic
Follow-up of #270.
We should use [`@interval`](https://juliaintervals.github.io/IntervalArithmetic.jl/latest/usage.html#Creating-intervals-1) instead of `IntervalArithmetic.Interval` to create intervals.

### PR DESCRIPTION
**Pull request by mforets**
Closes #300.

### CODE REVIEW

**Comment by mforets** [comment]
This section illustrates why before we didn't have problems in the tests and now i had to change them, since are using `@interval` instead of `IntervalArithmetic.Interval`.

```julia
julia> using IntervalArithmetic

julia> Interval(Float32(1.0), Float32(2.0))

[1, 2]

julia> typeof(ans)
IntervalArithmetic.Interval{Float32}

julia> @interval(Float32(1.0), Float32(2.0))
ERROR: MethodError: Cannot `convert` an object of type IntervalArithmetic.Interval{Float64} to an object of type Float32
This may have arisen from a call to the constructor Float32(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] Float32(::IntervalArithmetic.Interval{Float64}) at ./sysimg.jl:77

julia> @interval(1.0, 2.0)
[1, 2]

julia> typeof(ans)
IntervalArithmetic.Interval{Float64}

julia> setprecision(Interval, precision(Float32))
24

julia> @interval(1.0, 2.0)
[1, 2]₂₄

julia> typeof(ans)
IntervalArithmetic.Interval{BigFloat}
```

File: test/unit_overapproximate.jl [hunk 1]
```diff
@@ -90,10 +90,16 @@ for N in [Float64, Float32] # TODO Rational{Int}
...
 
     # Interval approximation
-    b = Ball1(N[0.], N(1.))
-    p = overapproximate(b, LazySets.Interval)
-    for d in to_N(N, [[1.], [-1.]])
-        @test σ(d, p)[1] ≈ σ(d, b)[1]
+    # for use with other precision, use
+    # setprecision(IntervalArithmetic.Interval, precision(N))
+    # however, this will use BigFloat's of given number of bits, which is a
+    # different numeric type than d, so this will triggeer a MethodError
```
Comment (schillic) [inline test/unit_overapproximate.jl]: "triggeer"

File: test/unit_decompose.jl [hunk 2]
```diff
@@ -40,9 +40,12 @@ for N in [Float64, Float32] # TODO Rational{Int}
     @test d.array[1] isa Hyperrectangle && test_directions(d.array[1])
     d = decompose(b, set_type=HPolygon, ɛ=to_N(N, 1e-2))
     @test d.array[1] isa HPolygon && test_directions(d.array[1])
+
+    if N == Float64
     d = decompose(b, set_type=LazySets.Interval, blocks=ones(Int, 6))
```
Comment (schillic) [inline test/unit_decompose.jl]: indent?

**Comment by schillic** [comment]
Actually, the idea of `@interval` is to create an interval from numerals (as in the tests). In the source code, it does not help because we already have floating point numbers. But let us keep it.

Does that mean that we cannot create `Interval`s of type `Rational{Int}` anymore? Can we provide constructors for different types at least? We should add a note to the docs of the `Interval` type.

:+1: for using `import`. This could (I did not try) have removed the warnings mentioned in #194.

**Comment by mforets** [comment]
> 👍 for using import. This could (I did not try) have removed the warnings mentioned in #194.

yes, i think so.

---

> Does that mean that we cannot create Intervals of type Rational{Int} anymore?

Hmmm...

```julia
julia> Interval(0, 1)
LazySets.Interval{Float64,IntervalArithmetic.Interval{Float64}}([0, 1])

julia> Interval(0//1, 2//1)
LazySets.Interval{Float64,IntervalArithmetic.Interval{Float64}}([0, 2])
```

This is not great. At least as you say we should provide constructors. I will ask `IntervalArithmetic` developers. (Since yesterday, there is `#intervals` in `https://julialang.slack.com/`).

File: test/unit_decompose.jl [hunk 3]
```diff
@@ -42,8 +41,8 @@ for N in [Float64, Float32] # TODO Rational{Int}
     @test d.array[1] isa HPolygon && test_directions(d.array[1])
 
     if N == Float64
```
Comment (schillic) [inline test/unit_decompose.jl]: Can be removed again, right?

**Pull request merged**